tempesta_fw/ss_skb.c

```diff
@@ -236,14 +236,11 @@ __it_next_data(struct sk_buff *skb, int i, TfwStr *it)
 static inline void
 __skb_insert_after(struct sk_buff *skb, struct sk_buff *nskb)
 {
-	SsSkbCb *scb = TFW_SKB_CB(skb);
-	SsSkbCb *nscb = TFW_SKB_CB(nskb);
-
-	nscb->next = scb->next;
-	nscb->prev = skb;
-	scb->next = nskb;
-	if (nscb->next)
-		TFW_SKB_CB(nscb->next)->prev = nskb;
+	nskb->next = skb->next;
+	nskb->prev = skb;
+	skb->next = nskb;
+	if (nskb->next)
+		nskb->next->prev = nskb;
 }
```

Comment:
The function must be removed since it does just the same as standard `list_add()` except it breaks the standard list behaviour using NULL as a tail pointer. The point of #166 is that our internal skb lists mut be the same as in the kernel.

Reply:
Our current internal implementation is based and largely tied with NULL-terminated lists, so we cannot use full circular lists without significant redesign of code which work  with our internal skb queue.